### PR TITLE
ddl: set tiflash placement group index to 120

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: ref #37171

### What is changed and how it works?

1. Set the group_index of placement rule group 'tiflash' to 120 when adding a tiflash placement rule.
2. Fix the group_index in `PollTiFlashRoutine` when the background task started for old cluster.

### Check List

Tests 

- [x] Unit test
- [x] Manual test (add detailed scripts or steps below)

Manual test1:
1. Create a table with tiflash replica in the old cluster.
2. Upgrade one tidb instance of the cluster to the code of this PR, the tiflash group_index was fixed to 120.

Manual test2:
1. Setup a new cluster with the code of this PR.
2. Create a new table with tiflash replica, the tiflash group_index is  120.

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by CalvinNeo — APPROVED** [review]
lgtm

File: ddl/ddl_tiflash_test.go [hunk 1]
```diff
@@ -995,3 +995,19 @@ func TestTiFlashBatchUnsupported(t *testing.T) {
...
+	timeout := time.Now().Add(10 * time.Second)
+	errMsg := "time out"
+	for time.Now().Before(timeout) {
+		time.Sleep(100 * time.Millisecond)
+		if s.tiflash.GroupIndex != 0 {
+			errMsg = "invalid group index"
+			break
+		}
+	}
+	require.Equal(t, uint64(120), s.tiflash.GroupIndex, errMsg)
```
Comment (bb7133) [inline ddl/ddl_tiflash_test.go:1012]: Change `120` to `placement.RuleIndexTiFlash`?

**Comment by bb7133** [comment]
> Fix the group_index in PollTiFlashRoutine when the background task started for old cluster.

@lcwangchao @CalvinNeo Will this change any other behaviors related to TiFlash replicas?

**Comment by lcwangchao** [comment]
> > Fix the group_index in PollTiFlashRoutine when the background task started for old cluster.
> 
> @lcwangchao @CalvinNeo Will this change any other behaviors related to TiFlash replicas?

I think it will have no other effects. @CalvinNeo Can you make a confirm again?

**Review by bb7133 — APPROVED** [review]
LGTM

File: domain/infosync/tiflash_manager.go [hunk 2]
```diff
@@ -69,8 +71,62 @@ func (m *TiFlashPDPlacementManager) Close(ctx context.Context) {
 
 }
 
+// SetTiFlashGroupConfig sets the tiflash's rule group config
+func (m *TiFlashPDPlacementManager) SetTiFlashGroupConfig(ctx context.Context) error {
```
Comment (xhebox) [inline domain/infosync/tiflash_manager.go:75]: It is fine to just override the old group config. There is no need to `get&put`. BTW, `override: true` is needed to make `Index` effective.

Comment (lcwangchao) [reply domain/infosync/tiflash_manager.go:75]: In the future the definition of `placement.RuleGroupConfig` changes, we can avoid modify other fields. We only need to set index and override fields.

**Review by xhebox — APPROVED** [review]

**Pull request merged**